src/main/java/com/avispl/symphony/dal/communicator/management/zoom/rooms/ZoomRoomsCommunicator.java

```diff
@@ -0,0 +1,395 @@
+     * @param command to perform
+     * @param verifyString string that should be present in a response payload
+     * @return boolean value indicating whether the operation is successful
+     */
+    private boolean executeAndVerify(String command, String verifyString) throws Exception {
+        String response = execute(command, verifyString);
+        return !StringUtils.isEmpty(response);
+    }
+
+    /**
+     * Execute the command and check the response to contain "verifyString" in a few attempts.
+     * It's not added into the general "CommandSuccess/Failure" list because the basic communicator
+     * checks for endsWith() but there's no guarantee that the response will end with a certain string,
+     * also sometimes there are multiple comands perfomed automatically on entry with multiple "** end" breaks.
+     * This part helps to have another layer of response verification.
+     *
+     * @param command to perform
+     * @param verifyString string that should be present in a response payload
+     * @return empty string or a command response, if successful
+     */
+    private String execute(String command, String verifyString) throws Exception {
+        String response;
+        controlOperationsLock.lock();
+        try {
+            updateLatestControlTimestamp();
+            refreshSshConnection();
+            response = send(command);
+            int retryAttempts = 0;
+            while (!response.contains(verifyString)){
+                retryAttempts++;
+                response = send(command);
+                if(retryAttempts >= 10) {
+                    return "";
+                }
+            }
+        } finally {
+            controlOperationsLock.unlock();
+        }
+        return response;
+    }
+
+    /**
+     * Check if the channel is connected and if not - reconnect
+     */
+    private void refreshSshConnection() throws Exception {
+        if (!isChannelConnected()) {
+            connect();
+        }
+    }
+
+    /**
+     * Update timestamp of the latest control operation
+     * */
+    private void updateLatestControlTimestamp(){
+        latestControlTimestamp = new Date().getTime();
+    }
+
+    /***
+     * Check whether the control operations cooldown has ended
+     * @return boolean value indicating whether the cooldown has ended or not
+     */
+    private boolean isValidControlCoolDown(){
+        return (new Date().getTime() - latestControlTimestamp) < controlsCooldownTimeout;
+    }
+
+    @Override
+    public String dial(DialDevice dialDevice) throws Exception {
+        String meetingStatus = getCallStatus();
+        if(meetingStatus.equals("in meeting")){
+            logger.warn("Not able to connect. Meeting " + meetingNumber + " is in progress.");
+            return null;
+        } else if (meetingStatus.equals("connecting meeting")) {
+            callDisconnect();
+        }
+        return joinStartMeeting(dialDevice.getDialString());

```

Comment:
Yeah it would be able to invite SIP/H323 after ZR instance is already associated with a specific meeting - https://devforum.zoom.us/t/zoomrooms-cli-calling-an-h-323-sip-device/31616

Reply:
@MaksimR-AVISPL In order to answer my question, please forget about zoom meetings for a moment. Please pretend that Zoom meetings do not exist, there is no such things in the world.
The dial() method of a codec interface is not about dialing to the meeting. It is dialing to any address provided. Thus my question again: Can ZoomRoom connect to anything else other than Zoom meeting? E.g. SIP address of another device? SIP address of any other (not Zoom) meeting? Like all other room devices (Cisco, Polycom, whoever) can do.

Reply:
Yes, that's what I meant - unfortunately, in order to connect to anything that is not Zoom meeting with SIP/H323 or something else using CLI - ZR instance has to be connected to a Zoom meeting first. The link above has a Zoom Staff' comment on this one

Reply:
Maksym, please understand that it is a business decision, not just a code. You are writing generic codec adapter which Symphony will use for any type of calls, not just zoom meetings. You need to understand how it will work and those cases and what will happen. E.g. if ZR is a part of point-to-point meeting with SIP/H.323 device, it will not be able to connect to the other party and such meeting will fail in Symphony. Or, if ZR is part of point-to-point meeting with SIP/H.323 device, it will create a Zoom Meeting and then invite that SIP/H.323 device into that meeting. Not sure how are you going to do that though in Symphony (what Zoom meeting will you use?), but, even to consider that possibility, you need to know that it works and what is needed for it to work.
Only then you can write proper code in your dial-in method, and, equally important, add appropriate release notes listing all the limitations (if any) adapter will have. As it stands now, both QA and Vince (and me), and feature customers assume that this adapter will be able to do all the type of meetings Symphony supports.

Reply:
I'm still not sure it is coded by decisions made after last review. Will it work if destination address is SIP or H.323 device?

Reply:
According to the latest request - regular SIP/H323 calls support is not implemented since SIP/H323 calls require strict separation between the Zoom SIP/H323 and regular ones. Added details to the javadocs.